src/iam.ts

```diff
@@ -1558,7 +1573,9 @@ export class IAM extends IAMBase {
             throw new Error(ERROR_MESSAGES.SIGNER_NOT_INITIALIZED);
         }
 
-        await this.verifyEnrolmentPrerequisites({ subject, role: claim.claimType });
+        const requiredRoles = await this.verifyEnrolmentPrerequisites({ subject, role: claim.claimType });
+        const requiredRoleDefinitions = await this.getRolesDefinition({ namespaces: requiredRoles });
+        await varifyRequiredParams({ claimParams: claim.claimParams, requiredRoleDefinitions });
```

Comment:
I'm a bit confused why this is change is being made. So I guess we were already checking first checking whether or not the subject had the prerequisite/required roles before issuing the claim to then. I think this makes sense as maybe they snuck in a claim request without having the necessary prerequisites.

But then you're adding an additional check that the `claimParams` that I guess were added by the issuer are required by the enrolmentPrequisite roles? I don't think this is right. I think the check should be of the `claimParams` against some `issuerFields` in the requested role metadata.

Sorry, not sure if this is ready for review of not, but thought I would check in

Also, minor typo:
```suggestion
        const requiredRoles = await this.verifyEnrolmentPrerequisites({ subject, role: claim.claimType });
        const requiredRoleDefinitions = await this.getRolesDefinition({ namespaces: requiredRoles });
        await verifyRequiredParams({ claimParams: claim.claimParams, requiredRoleDefinitions });
```

Reply:
typo fixed. And as for preconditions - yes , I misunderstood the concept. I changed verification of required params from prerequisites to issued role.